secret/scanner_test.go

```diff
@@ -0,0 +1,116 @@
+package secret_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aquasecurity/fanal/secret"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func TestSecretScanner(t *testing.T) {
+	wantFinding1 := types.SecretFinding{
+		RuleID:    "rule1",
+		Category:  "general",
+		Title:     "Generic Rule",
+		Severity:  "HIGH",
+		StartLine: 2,
+		EndLine:   3,
+		Match:     "generic secret line *****",
+	}
+	wantFinding2 := types.SecretFinding{
+		RuleID:    "rule1",
+		Category:  "general",
+		Title:     "Generic Rule",
+		Severity:  "HIGH",
+		StartLine: 4,
+		EndLine:   5,
+		Match:     "*****",
+	}
+	tests := []struct {
+		name          string
+		configPath    string
+		inputFilePath string
+		want          types.Secret
+	}{
+		{
+			name:          "find match",
+			configPath:    "testdata/config.yaml",
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{wantFinding1, wantFinding2},
+			},
+		},
+		{
+			name:          "allow-rule path",
+			configPath:    "testdata/allow-path.yaml",
+			inputFilePath: "testdata/secret.txt",
+			want:          types.Secret{},
+		},
+		{
+			name:          "allow-rule regex",
+			configPath:    "testdata/allow-regex.yaml",
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{wantFinding1},
+			},
+		},
+		{
+			name:          "exclude-block regexes",
+			configPath:    "testdata/exclude-block.yaml",
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{wantFinding2},
+			},
+		},
+		{
+			name:          "global allow-rule path",
+			configPath:    "testdata/global-allow-path.yaml",
```

Comment:
Do you think we should add another test to check that the global allow path works? 
Right now it checks that paths that shouldn't be allowed, are not allowed. But it doesn't cover the case if the feature is broken.

Reply:
Do you mean it doesn't cover the case when the scanning is completely broken and doesn't detect anything?

Reply:
I mean that it doesn't check that the `global allow-rule path` skips any secrets.
The path `foo` doesn't exist so all secrets are detected, but at the same time, the `global allow-rule path` could not work and it will detect all the secrets as well.

Reply:
Cool, I'll fix that.